nixos/modules/services/misc/redmine.nix

```diff
@@ -252,28 +266,28 @@ in
         subversion
       ];
       preStart = ''
-        # ensure cache directory exists for db:migrate command
-        mkdir -p "${cfg.stateDir}/cache"
+        rm -rf "${cfg.stateDir}/plugins"
+        rm -rf "${cfg.stateDir}/public/themes"
 
         # create the basic directory layout the redmine package expects
-        mkdir -p /run/redmine/public
-
-        for i in config files log plugins tmp; do
-          mkdir -p "${cfg.stateDir}/$i"
-          ln -fs "${cfg.stateDir}/$i" /run/redmine/
-        done
-
-        for i in plugin_assets themes; do
-          mkdir -p "${cfg.stateDir}/public/$i"
-          ln -fs "${cfg.stateDir}/public/$i" /run/redmine/public/
-        done
-
+        mkdir -m 0750 -p \
```

Comment:
can't we create these directories through `systemd.tmpfiles.rules` too?

Reply:
No need. The root has already been created with appropriate permission so the preStart script is free to operate here.

Do you have a style preference that says use tmpfiles.d over bash?

Reply:
I do prefer declaring expected directory structure in systemd notation instead of shell scripts setting it up (so [`StateDirectory=, CacheDirectory=, LogsDirectory=, and ConfigurationDirectory= ` over] `tmpfiles.d` over `bash)` - but that's more personal preference than general consensus.

Looking deeper into the setup script, do we really need to set all this up at each start, or can't we pre-build parts of the structure through nix?

Reply:
@flokli That makes sense. I'll rework this to use tmpfiles.d and get back to you.

Reply:
@flokli I think this should be good enough. Look alright to you?